### PR TITLE
Cooking-UseCase | Beta version 1

### PR DESCRIPTION
**Pull request by Ovakefali13**
The first working version of the UseCase
Implemented:
- All methods of the sequence diagram
- Integration tests (all positive MOCK and DONOTMOCK)

Missing:
- Regex for Chatbot
- [x] advance-method

### CODE REVIEW

**Comment by mariushegele** [comment]
#70

**Comment by mariushegele** [comment]
#69 If possible implement the Usecase interface (inherit from Usecase)

**Comment by Ovakefali13** [comment]
@mariushegele Is that what you had in mind ?

**Comment by Ovakefali13** [comment]
Added `advance` and `is_finished` needs aproval from @mariushegele

File: usecase/cooking/test/test_usecase.py [hunk 1]
```diff
@@ -25,15 +30,27 @@ def setUpClass(self):
...
+            self.use_case.todoist_service = self.todoist_service
 
-    def test_trigger_usecase(self):
-        self.use_case.trigger_use_case('pork')
+            self.spoonacle_service = SpoonacularService.instance()
+            self.spoonacle_service.set_remote(SpoonacularMOCKRemote())
+            self.use_case.spoonacle_service = self.spoonacle_service
+            
+    def test_usecase(self):
+        self.use_case.advance({'ingredient': 'pork'})
```
Comment (mariushegele) [inline usecase/cooking/test/test_usecase.py]: why don't you take the return of `advance` as a reply?

you could insert an assertIsInstance(reply, Reply)

are there any other asserts that could validate the correct behavior of the test? E.g. a certain substring that is contained in the reply?

Comment (Ovakefali13) [reply usecase/cooking/test/test_usecase.py]: done

**Pull request merged**